### PR TITLE
Record move sequence

### PR DESCRIPTION
**Pull request by kaiweike**
## Description
This pull request introduces modifications to enable recording and outputting the sequence of moves.

## Changes
- Implemented `record_move` function to track each move.
- Implemented `print_moves` function to output the sequence of moves.
- Initialized `best_move` to -1 in `negamax` function. This serves as a flag to determine whether a move should be recorded.

### CODE REVIEW

File: ttt.c [hunk 1]
```diff
@@ -3,6 +3,30 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define MAX_MOVES 9
```
Comment (jserv) [inline ttt.c]: Avoid hardcoded literal. Specify `(3 * 3)` instead.

**Review by jserv — CHANGES_REQUESTED** [review]
Check https://cbea.ms/git-commit/ carefully and rework git commit messages.

File: ttt.c (hunk 1 above)
Comment (kaiweike) [reply ttt.c]: [done](https://github.com/kaiweike/ttt/commit/3384fcf9f2d71ec64ef1c91d947cfc6812434158)

**Comment by kaiweike** [comment]
Thanks for the suggestions. I have made the following changes:
- Updated the definition of `MAX_MOVES` to avoid hardcoded literal.
- Reworked the git commit messages.

**Comment by jserv** [comment]
Thank @kaiweike for contributing!

**Pull request merged**